core/common/file.hpp

```diff
@@ -24,7 +24,8 @@ namespace fc::common {
 
   /** returns true on success */
   inline bool read(std::istream &is, gsl::span<uint8_t> bytes) {
-    return is.read((char *)bytes.data(), bytes.size()).good();
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    return is.read(reinterpret_cast<char *>(bytes.data()), bytes.size()).good();
```

Comment:
may use "bytestr" or "cast" from "common/span.hpp", which already have this "nolint"